js/form.js

```diff
@@ -188,15 +230,29 @@
       return buckets;
     }
 
-    return {
-      buckets: extractBuckets(getHighestCount(fromPetition.data.attributes.signatures_by_constituency)),
-      bucketFor: function(count) {
-        for(var i = 0; i < 8; i++) {
-          if (count <= this.buckets[i]) {
-            return i;
+    if (PetitionMap.is_weighted) {
+      return {
+        buckets: extractBuckets(getHighestCount(fromPetition)),
+        bucketFor: function(count) {
+          for(var i = 0; i < 8; i++) {
+            if (count <= this.buckets[i]) {
+              return i;
+            }
+          }
+          return 8;
+        }
+      }
+    } else {
+        return {
+        buckets: extractBuckets(getHighestCount(fromPetition.data.attributes.signatures_by_constituency)),
+        bucketFor: function(count) {
+          for(var i = 0; i < 8; i++) {
+            if (count <= this.buckets[i]) {
+              return i;
+            }
           }
+          return 8;
```

Comment:
`bucketFor` is the same in both cases isn't it?  We could put the conditional around the value we set for `buckets` to make that clearer.